src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,202 @@
+NM_CONNECTION_DISCONNECTED = 2
+
+class DeviceNotFoundException(Exception):
+    """Raised when an interface name passed to NetworkManagerHelper is not found."""
+
+class NetworkManagerHelper:
+    """NetworkManagerHelper abstracts away some of the messy details of the NetworkManager
+    Dbus API.
+    """
+
+    def __init__(self, config):
+
+        self.logger = logging.getLogger(__name__)
+        self.network_activation_timeout = config['network_activation_timeout']
+        self.wired_network_name = config['wired_network_name']
+
+        self.network_id_table = self._build_network_id_table()
+        self._get_device_objects(config['wired_interface_name'],
+            config['wifi_interface_name'])
+
+    def activate_network(self, network_id):
+        """Tells NetworkManager to activate a network with the supplied network_id.
+        Returns True if there are no errors, False otherwise.
+        """
+        success = False
+        connection = self.network_id_table[network_id]
+        network_device = self._get_device_for_connection(connection)
+
+        networkmanager_output = NetworkManager.NetworkManager.ActivateConnection(connection, network_device, '/')
+        self._run_proxy_call(networkmanager_output)
+        success = self._wait_for_connection(connection)
+
+        return success
+
+    def get_network_ip(self, network_id):
+        """Attempts to retrieve the ip address associated with the given network id. If it
+        is unable to, it returns None."""
+        # I decided not to throw an exception here because the intended caller would end up
+        #   simply using it for flow control.
+        ip_address = None
+
+        connection = self.network_id_table[network_id]
+
+        if self._wait_for_connection(connection):
+            device = self._get_device_for_connection(connection)
+            ip_address = device.Ip4Config.AddressData[0]['address']
+        else:
+            self.logger.warning('Attempted to get IP address for network that is not connected.')
+
+        return ip_address
+
+    def network_is_ready(self, network_id):
+        """Check whether the network with the given network id is ready."""
+        connection = self.network_id_table[network_id]
+        return self._wait_for_connection(connection)
+
+    def _build_network_id_table(self):
+        """Assemble a helpful dictionary of network objects, indexed by the connection's
+        id in NetworkManager.
+        """
+        network_id_table = {}
+
+        all_connections = NetworkManager.Settings.ListConnections()
+        self._run_proxy_call(all_connections)
+
+        for connection in all_connections:
+            connection_id = connection.GetSettings()['connection']['id']
+            network_id_table[connection_id] = connection
+
+        return network_id_table
+
+    def _get_active_connection(self, network_id):
+        """Returns the active connection object associated with the given network id."""
```

Comment:
Again, I'd explicitly describe the parameter and the value returned in a manner similar to what I did in gpgmailer.